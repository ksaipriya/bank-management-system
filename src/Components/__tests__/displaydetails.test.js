import React from 'react';
import { render } from '@testing-library/react';
import LoanDetails from '../displaydetails.js'
import Adapter from 'enzyme-adapter-react-16';
import { configure, ShallowWrapper, shallow, mount } from 'enzyme';
import ShallowRenderer from 'react-test-renderer/shallow';

configure({ adapter: new Adapter() });
describe('Test case for testing display details component', () => {

    it('matches snapshot', () => {
        const shallowRenderer = new ShallowRenderer();
        shallowRenderer.render(<LoanDetails />);
        const result = shallowRenderer.getRenderOutput();
        expect(result).toMatchSnapshot();
    });

    it('able to find an html element', () => {
        let shallowWrapper = new ShallowWrapper(<LoanDetails />)
        expect(shallowWrapper.find('h4').html()).toContain('You have successfully applied for a loan, here are you loan details');
    });

    it('header check', () => {
        let shallowWrapper = new ShallowWrapper(<LoanDetails />)
        expect(shallowWrapper.find('b').html()).toContain('Loan Details');
    });

    it('Label for Update details button', () => {
        let shallowWrapper = new ShallowWrapper(<LoanDetails />)
        expect(shallowWrapper.find('#update').html()).toContain('Update your personal details?')
    })

    it('Label for Logout button', () => {
        let shallowWrapper = new ShallowWrapper(<LoanDetails />)
        expect(shallowWrapper.find('#logout').html()).toContain('No more changes?')
    })

    it('Test Update Button', () => {
        const mockCallBack = jest.fn();
        const button = shallow(<button onClick={mockCallBack}>Update Details</button>);
        button.find('button').simulate('click');
        expect(mockCallBack.mock.calls.length).toEqual(1);
    });

    it('Test Logout Button', () => {
        const mockCallBack = jest.fn();
        const button = shallow(<button onClick={mockCallBack}>Logout</button>);
        button.find('button').simulate('click');
        expect(mockCallBack.mock.calls.length).toEqual(1);
    });

    it('Test Loan type field', () => {
        const a = <td>Loan Type</td>
        let shallowWrapper = new ShallowWrapper(<LoanDetails />)
        expect(shallowWrapper.find('td').getElements()[0]).toMatchObject(a)
    })

    it('Test Loan Amount field', () => {
        const a = <td>Loan Amount</td>
        let shallowWrapper = new ShallowWrapper(<LoanDetails />)
        expect(shallowWrapper.find('td').getElements()[2]).toMatchObject(a)
    })

    it('Test Loan Apply Date field', () => {
        const a = <td>Loan Apply Date</td>
        let shallowWrapper = new ShallowWrapper(<LoanDetails />)
        expect(shallowWrapper.find('td').getElements()[4]).toMatchObject(a)
    })

    it('Test Loan Issue Date field', () => {
        const a = <td>Loan Issue Date</td>
        let shallowWrapper = new ShallowWrapper(<LoanDetails />)
        expect(shallowWrapper.find('td').getElements()[6]).toMatchObject(a)
    })

    it('Test Rate of Interest field', () => {
        const a = <td>Rate of Interest</td>
        let shallowWrapper = new ShallowWrapper(<LoanDetails />)
        expect(shallowWrapper.find('td').getElements()[8]).toMatchObject(a)
    })

    it('Test Duration of Loan field', () => {
        const a = <td>Duration of Loan</td>
        let shallowWrapper = new ShallowWrapper(<LoanDetails />)
        expect(shallowWrapper.find('td').getElements()[10]).toMatchObject(a)
    })
})