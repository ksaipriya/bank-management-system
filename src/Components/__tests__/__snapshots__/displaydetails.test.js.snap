// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test case for testing display details component matches snapshot 1`] = `
<Container
  fluid={false}
>
  <h4
    className="card-title"
  >
    You have successfully applied for a loan, here are you loan details
  </h4>
  <ForwardRef
    bordered={true}
    hover={true}
    striped={true}
  >
    <thead>
      <b
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Loan Details
      </b>
    </thead>
    <tbody>
      <tr>
        <td>
          Loan Type
        </td>
        <td />
      </tr>
      <tr>
        <td>
          Loan Amount
        </td>
        <td />
      </tr>
      <tr>
        <td>
          Loan Apply Date
        </td>
        <td />
      </tr>
      <tr>
        <td>
          Loan Issue Date
        </td>
        <td />
      </tr>
      <tr>
        <td>
          Rate of Interest
        </td>
        <td />
      </tr>
      <tr>
        <td>
          Duration of Loan
        </td>
        <td>
           years
        </td>
      </tr>
    </tbody>
  </ForwardRef>
  <Form
    inline={false}
  >
    <FormRow>
      <FormGroup
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        className="register"
      >
        <FormLabel
          column={false}
          id="update"
          srOnly={false}
        >
          Update your personal details?
        </FormLabel>
        <br />
        <Button
          active={false}
          disabled={false}
          onClick={[Function]}
          variant="success"
        >
          Update Details
        </Button>
      </FormGroup>
      <FormGroup
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          }
        }
        className="login"
      >
        <FormLabel
          column={false}
          id="logout"
          srOnly={false}
        >
          No more changes?
        </FormLabel>
        <br />
        <FormRow>
          <FormGroup
            as={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              }
            }
          >
            <Button
              active={false}
              disabled={false}
              onClick={[Function]}
              variant="primary"
            >
              Logout
            </Button>
          </FormGroup>
        </FormRow>
      </FormGroup>
    </FormRow>
  </Form>
</Container>
`;
